caffe2/operators/collect_and_distribute_fpn_rpn_proposals_op.cc

```diff
@@ -58,12 +58,11 @@ void SortAndLimitRoIsByScores(Eigen::Ref<const EArrXf> scores, int n,
   // Note that people have found nth_element + sort to be much faster
   // than partial_sort so we use it here
   if (n > 0 && n < rois.rows()) {
-    std::nth_element(idxs.begin(), idxs.begin() + n, idxs.end(), comp);
```

Comment:
Why did we remove this?

Reply:
I agree - we should keep this. The comment I made above about nth_element + sort hasn't really been tested on real-world cases, so it could likely be wrong. If you could verify given the larger range of test cases below, that would be great. Then we can decide whether to use nth_element + sort or partial_sort.

In both cases, though, we can make it stable by adjusting comp to also test index:

```
auto comp = [&scores](int lhs, int rhs) {
  if (scores(lhs) > scores(rhs)) return true;
  if (scores(lhs) < scores(rhs)) return false;
  return lhs > rhs;
};
```
Then there's no need to use stable_sort. Thoughts?